stan/math/prim/meta/apply_scalar_binary.hpp

```diff
@@ -0,0 +1,326 @@
+ * return scalar types differ (e.g., functions implicitly promoting
+ * integers).
+ */
+template <typename T1, typename T2>
+struct apply_scalar_binary<T1, T2,
+                           require_all_std_vector_vt<is_container, T1, T2>> {
+  /**
+   * Member function for applying a functor to two std::vectors of containers,
+   * and subsequently returning an std::vector of containers.
+   *
+   * @tparam T1 Type of first std::vector to which functor is applied.
+   * @tparam T2 Type of second std::vector to which functor is applied.
+   * @tparam F Type of functor to apply.
+   * @param x First std::vector input to which operation is applied.
+   * @param y Second std::vector input to which operation is applied.
+   * @param f functor to apply to std::vector inputs.
+   * @return std::vector with result of applying functor to inputs.
+   */
+  template <typename F>
+  static inline auto apply(const T1& x, const T2& y, const F& f) {
+    using T1_vt = value_type_t<T1>;
+    using T2_vt = value_type_t<T2>;
+    using T_return
+        = decltype(apply_scalar_binary<T1_vt, T2_vt>::apply(x[0], y[0], f));
+    size_t y_size = y.size();
+    std::vector<T_return> result(y_size);
+    for (size_t i = 0; i < y_size; ++i)
+      result[i] = apply_scalar_binary<T1_vt, T2_vt>::apply(x[i], y[i], f);
+    return result;
+  }
+};
+
+/**
+ * Specialisation for use with one nested container (std::vectors) and one
+ * scalar. The returned scalar type is deduced to allow for cases where the
+ * input and return scalar types differ (e.g., functions implicitly promoting
+ * integers).
+ */
+template <typename T1, typename T2>
+struct apply_scalar_binary<T1, T2,
+                           require_any_std_vector_vt<is_container, T1, T2>,
+                           require_any_stan_scalar_t<T1, T2>> {
+  /**
+   * Member function for applying a functor to an std::vector of containers
+   * and a scalar, and subsequently returning an std::vector of containers.
+   *
+   * @tparam T1 Type of std::vector to which functor is applied.
+   * @tparam T2 Type of scalar to which functor is applied.
+   * @tparam F Type of functor to apply.
+   * @param x std::vector input to which operation is applied.
+   * @param y Scalar input to which operation is applied.
+   * @param f functor to apply to inputs.
+   * @return std::vector with result of applying functor to inputs.
+   */
+  template <typename F, typename T1_2 = T1, typename T2_2 = T2,
+            require_std_vector_vt<is_container, T1_2>* = nullptr,
+            require_stan_scalar_t<T2_2>* = nullptr>
+  static inline auto apply(const T1& x, const T2& y, const F& f) {
+    using T1_vt = value_type_t<T1>;
+    using T_return
+        = decltype(apply_scalar_binary<T1_vt, T2>::apply(x[0], y, f));
+    size_t x_size = x.size();
+    std::vector<T_return> result(x_size);
+    for (size_t i = 0; i < x_size; ++i)
+      result[i] = apply_scalar_binary<T1_vt, T2>::apply(x[i], y, f);
```

Comment:
Please use braces consistently, even if a body of a for/if is just one line.

Reply:
Fixed